Feature/dark mode

fryl0ch opened the pull request:
I took a stab at implementing dark mode as requested in #13

joebeachjoebeach commented on index.html:
```diff
@@ -402,6 +408,56 @@ <h2 class="title-bar-item">JSON Path Finder</h2>
+});
+
+function invert() { // dark mode
+  let css = 'html {-webkit-filter: invert(100%) brightness(.37) saturate(2.8) hue-rotate(180deg);' +
```
I think we need a more accessible contrast and I'd prefer a softer black. Fiddling around, this feels nice:
```suggestion
  let css = 'html {-webkit-filter: invert(85%) brightness(.90) saturate(1.5) hue-rotate(180deg);' +
```

joebeachjoebeach commented on index.html:
```diff
@@ -37,6 +39,7 @@ <h2 class="title-bar-item">JSON Path Finder</h2>
         <button class="editor-button-row-button" @click="loadSample">Sample</button>
         <button class="editor-button-row-button" @click="beautify">Beautify</button>
         <button class="editor-button-row-button" @click="minify">Minify</button>
+        <button class="editor-button-row-button" @click="darkmode">Dark Mode</button>
```
Two requests:

1. Could you place the button either in the header to the left of the github logo/link, or on the far right side of the footer? Flat button with the same background color as the header/footer.
2. Can you have the text toggle between `Light Mode` and `Dark Mode` depending on which mode is active? I _think_ that should be doable with Vue's templating, but the last time I used Vue was when I made this 6 years ago... No stress if it's a big lift.

joebeachjoebeach commented on index.html:
```diff
@@ -385,6 +388,9 @@ <h2 class="title-bar-item">JSON Path Finder</h2>
     minify: function() {
       this.editor.setValue(vkbeautify.jsonmin(this.editor.getValue()), 1);
+    },
+    darkmode: function() {
```
naming nit/request
```suggestion
    toggleDarkmode: function() {
```

joebeachjoebeach reviewed (commented):
Thank you for this! I have some requests but on this whole this is great.

fryl0ch commented:
I think these changes should address your review.  I tried sticking the button in the footer but had trouble with the css, so I put it in the header with a round image to match the GitHub logo. Alt text of the link updates with mode changes.

joebeachjoebeach commented on index.html:
```diff
@@ -27,6 +29,8 @@
         <h2 class="title-bar-item">JSON Path Finder</h2>
       </div>
       <div class="title-bar-item">
+        <a href="#" @click="toggleDarkMode" v-bind:alt='(isDarkMode ? "Light Mode" : "Dark Mode")'><img width="30" height="30" src="./images/contrast-icon.png"></a>
```
this will need to be a `button` as links are for navigation. you can add an id to it and add a corresponding selector to the css file to style it to undo the default button styles I put in place. since `alt` won't be available to us anymore on the button, we could use `alt` on the image I guess, plus `title` for a tooltip on it.
```css
#dark-mode-button {
  height: auto;
  width: auto;
  background-color: transparent;
}
```
when modifying `css/style.20210306.css`, you'll need to rename it, replacing the timestamp with the current date in YYYYMMDD format to bust browser+cdn caches.

joebeachjoebeach commented on index.html:
```diff
@@ -397,11 +406,60 @@ <h2 class="title-bar-item">JSON Path Finder</h2>
 
+window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', event => {
+    const newColorScheme = event.matches ? "dark" : "light";
+    console.log(newColorScheme);
```
```suggestion
```

joebeachjoebeach reviewed (commented):
thanks for the quick turnaround. just two comments

fryl0ch commented:
I switched the <a> to a <button> and added the style to the css. 

for some reason safari wasn't picking up the new style until I shoved it into an inline style attribute. less than ideal, but it works :P

joebeachjoebeach commented:
Thanks!
